soh/soh/Enhancements/sfx-editor/SfxEditor.cpp

```diff
@@ -0,0 +1,487 @@
+    { NA_SE_EN_GOMA_JR_CRY, { "Gohma Larva Croak", "0x395D", SFX_SOUND } },
+    { NA_SE_EN_GOLON_WAKE_UP, { "Goron Wake", "0x38FC", SFX_SOUND } },
+    { NA_SE_SY_START_SHOT, { "Gunshot", "0x4835", SFX_SOUND } },
+    { NA_SE_IT_HAMMER_HIT, { "Hammer Bonk", "0x180A", SFX_SOUND } },
+    { NA_SE_EN_IRONNACK_SWING_AXE, { "Iron Knuckle", "0x3929", SFX_SOUND } },
+    { NA_SE_EN_FANTOM_ST_LAUGH, { "Phantom Ganon Laugh", "0x39D6", SFX_SOUND } },
+    { NA_SE_EV_PLANT_BROKEN, { "Plant Explode", "0x284E", SFX_SOUND } },
+    { NA_SE_SY_GET_RUPY, { "Rupee", "0x4803", SFX_SOUND } }, 
+    { NA_SE_VO_RT_CRASH, { "Ruto Crash", "0x6860", SFX_SOUND } },
+    { NA_SE_VO_RT_LIFT, { "Ruto Lift", "0x6864", SFX_SOUND } },
+    { NA_SE_VO_RT_THROW, { "Ruto Thrown", "0x6865", SFX_SOUND } },
+    { NA_SE_EN_NUTS_UP, { "Scrub Emerge", "0x387C", SFX_SOUND } },
+    { NA_SE_EN_AWA_BOUND, { "Shabom Bounce", "0x3948", SFX_SOUND } },
+    { NA_SE_EN_SHELL_MOUTH, { "Shellblade", "0x3849", SFX_SOUND } },
+    { NA_SE_EN_STALTU_DAMAGE, { "Skulltula Damage", "0x386B", SFX_SOUND } },
+    { NA_SE_EN_NUTS_THROW, { "Spit Nut", "0x387E", SFX_SOUND } },
+    { NA_SE_VO_TA_CRY_0, { "Talon Hmm", "0x6852", SFX_SOUND } },
+    { NA_SE_VO_TA_SURPRISE, { "Talon Surprised", "0x6851", SFX_SOUND } },
+    { NA_SE_SY_LOCK_ON, { "Target Enemy", "0x4830", SFX_SOUND } },
+    { NA_SE_SY_LOCK_ON_HUMAN, { "Target Neutral", "0x4810", SFX_SOUND } }
+};
+
+const std::map<u16, const std::string> ocarinaSFXMap = { { 1, "Ocarina" }, { 2, "Voice" },     { 3, "Whistle" },
+                                                         { 4, "Harp" },    { 5, "MusicBox"}, { 6, "Flute" } };
+
+void Draw_BgmTab(const std::string& tabKey, const std::map<u16, std::tuple<std::string, std::string, SeqType>>& map,
+                 SeqType type) {
+    srand(time(NULL));
+    if (ImGui::Button("Reset All")) {
+        for (const auto& [defaultValue, nameAndStorageKeySuffix] : map) {
+            const auto& [name, storageKeySuffix, seqType] = nameAndStorageKeySuffix;
+            if (seqType == type) {
+                const std::string storageKey = "gSfxEditor_" + storageKeySuffix;
+                CVar_SetS32(storageKey.c_str(), defaultValue);
+            }
+        }
+        SohImGui::needs_save = true;
+    }
+    ImGui::SameLine();
+    if (ImGui::Button("Randomize All")) {
+        std::vector<u16> values;
+        for (const auto& [value, nameAndStorageKeySuffix] : map) {
+            const auto& [name, storageKeySuffix, seqType] = nameAndStorageKeySuffix;
+
+            if (seqType == type) {
+                values.push_back(value);
+            }
+        }
+        for (const auto& [defaultValue, nameAndStorageKeySuffix] : map) {
+            const auto& [name, storageKeySuffix, seqType] = nameAndStorageKeySuffix;
+            const std::string storageKey = "gSfxEditor_" + storageKeySuffix;
+            if (seqType == type) {
+                const int randomIndex = rand() % values.size();
```

Comment:
I have work planned to remove all uses of C's rand() function.  Instead, you should be calling 3D rando's Random function:  https://github.com/HarbourMasters/Shipwright/blob/develop-zhora/soh/soh/Enhancements/randomizer/3drando/random.cpp#L15.